caikit_nlp/data_model/embedding_vectors.py

```diff
@@ -0,0 +1,188 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Data structures for embedding vector representations
+"""
+
+# Standard
+from typing import List, Union
+import json
+
+# Third Party
+from py_to_proto.dataclass_to_proto import Annotated, FieldNumber, OneofField
+import numpy as np
+
+# First Party
+from caikit.core import DataObjectBase, dataobject
+from caikit.core.toolkit import error_handler
+import alog
+
+log = alog.use_channel("DATAM")
+error = error_handler.get(log)
+
+
+@dataobject(package="caikit_data_model.caikit_nlp")
+class FloatValue(DataObjectBase):
+    float_value: Union[
+        Annotated[float, OneofField("data_float"), FieldNumber(1)],
+        Annotated[np.float32, OneofField("data_float32"), FieldNumber(2)],
+        Annotated[np.float64, OneofField("data_float64"), FieldNumber(3)],
+    ]
+
+
+@dataobject(package="caikit_data_model.caikit_nlp")
+class Vector1D(DataObjectBase):
+    """Data representation for a 1 dimension vector.
+    NOTE: This only support float types
+    """
+
+    data: List[FloatValue]
```

Comment:
I think what you're trying to accomplish here is a representation of a consecutive sequence of numeric values backed by _some_ type of in-memory floating-point representation. As written, I think this is not the best way to represent that because as written, each element of the `List` can be a different in-memory type AND the `oneof` information needs to be held for each element of the list. I think a cleaner way to do this would be something like:

```py
@dataobject(...)
class Vector1D(DataObjectBase):

    @dataobject
    PyFloatSequence(DataObjectBase):
        values: List[float]

    @dataobject
    NpFloat32Sequence(DataObjectBase):
        values: List[np.float32]

    @dataobject
    NpFloat64Sequence(DataObjectBase):
        values: List[np.float64]

    data: Union[
        PyFloatSequence,
        NpFloat32Sequence,
        NpFloat64Sequence,
    ]
```

Reply:
Good point. I'll make the change